src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp

```diff
@@ -79,7 +79,10 @@ void addONNXToZHighPasses(
   pm.addNestedPass<func::FuncOp>(onnx_mlir::createShapeInferencePass());
   // There are more opportunities for const propagation once all zhigh ops were
   // generated.
-  pm.addNestedPass<func::FuncOp>(onnx_mlir::createConstPropONNXToONNXPass());
+  if ((/*enableConstantProp*/ optLevel >= 3 && !enableConstantProp) ||
```

Comment:
Nit:
```
if (/*enableConstantProp*/ optLevel >= 3 || enableConstantProp) 
```
is sufficient and easier to read. Please change here and where this pattern repeat.

Reply:
That will not cover all paths... I am happy to talk if need be.

Reply:
This simplified condition does not allow to manually disable constant propagation at -O3.

Reply:
For instance if we were to apply the simplification that means users are no longer forced to use content propagation at Level 03.... they are now able to turn it off if they want to. Please look at the explanation below:

```
meganhampton@Megans-MacBook-Pro-2 build % ./Debug/bin/onnx-mlir --O1 --EmitLib roberta-base-11.onnx --onnx-op-stats TXT --enable-constant-prop=false --shapeInformation 0:1x1 -v 
The ONNX model has 124645716 elements in 
```

This will allow a user to disable constant prorogation instead of enable.

Reply:
I must be missing something:

```
-O3 -enable=false  -> (O3 && !enable) || enable = true  || false = true
-O3 -enable=true   -> (O3 && !enable) || enable = false || true = true
-O0 -enable=false  -> (O3 && !enable) || enable = false || false = false
-O0 -enable=true   -> (O3 && !enable) || enable = false || true = true
```

vs 

```
-O3 -enable=false  -> O3 || enable = true
-O3 -enable=true   -> O3 || enable = true
-O0 -enable=false  -> O3 || enable = false
-O0 -enable=true   -> O3 || enable = true
```

Please let me know where I went wrong @tungld @hamptonm1 .

I do find it a bit confusing that setting explicitly `-enable-constant-prop=false` will still result in constant prop to be run. Maybe a better name is `force-constant-prop`.

Reply:
I agree with that @tungld , so we need to have an optional; if `enable` is given by user, use that. Otherwise, use the -O3.

Can you give an example of a boolean option that is has no default, and how to test if the option is defined, and whether its true or false?